VideoApp/VideoApp/Stores/Auth/CommunityAuthStore.swift

```diff
@@ -38,15 +38,16 @@ class CommunityAuthStore: AuthStoreWriting {
     func start() {
         guard let passcode = keychainStore.passcode else { return }
         
-        configureAPI(passcode: passcode)
+        try? configureAPI(passcode: passcode)
     }
 
     func signIn(email: String, password: String, completion: @escaping (AuthError?) -> Void) {
         fatalError("Email sign in not supported by community auth.")
     }
 
     func signIn(userIdentity: String, passcode: String, completion: @escaping (AuthError?) -> Void) {
-        configureAPI(passcode: passcode)
+        guard (try? configureAPI(passcode: passcode)) != nil else { completion(.passcodeIncorrect); return }
```

Comment:
It's kind of interesting to convert synchronous errors that can be thrown using fancy Swift error handling to more basic error handling required for asynchronous calls. For now to limit scope I just created a completely new error.

In the future I think I should move towards using `Error` in interfaces like this. This would make it much easier to pass around errors. For example if an error is thrown when I do `try` here the compiler only gives me an `Error`. So then I have to cast to `AuthError` just to forward it which is a pain. I think I should only have to cast when I need to add specific error handling and not just forwarding it. But I wanted to think about this more and also didn't want to add a lot of general error changes to this PR.

I used throw for the passcode decoding instead of optionals and booleans because it made the code quite a bit cleaner.